src/main/java/healthy/lifestyle/backend/workout/service/WorkoutServiceImpl.java

```diff
@@ -58,4 +58,34 @@ public List<WorkoutResponseDto> getDefaultWorkouts(String sortFieldName) {
                 })
                 .toList();
     }
+
+    @Transactional
+    @Override
+    public WorkoutResponseDto getWorkoutById(long id) {
+        WorkoutResponseDto workoutDto = modelMapper.map(workoutRepository.findById(id), WorkoutResponseDto.class);
```

Comment:
Let's throw an error if there is no workout found, and when requested workout is custom.

Consider adding a few negative test cases in unit and controller tests (if there is no workout with requested id, and requested id is protected).